src/runtime/statistics.py

```diff
@@ -220,6 +220,39 @@ def harmonic_mean(x: list[f64]) -> f64:
 
     return k / sum
 
+
+def mode(x: list[i32]) -> i32:
+    k: i32 = len(x)
+    y: list[i32]
+    y = deepcopy(x)
+    c: i32; i: i32; j: i32;
+    for i in range(k):
+        for j in range(0, k-1):
+            if y[j] > y[j+1]:
+                c = y[j]
+                y[j] = y[j+1]
+                y[j+1] = c
```

Comment:
This is a sort operation right?

Reply:
Why are we doing this if we are going to use `dict` anyways for storing value to frequency?

Reply:
Is this to preserve consistency of outputs because `dict` is a unordered?

Reply:
Oh, I changed the algorithm. It was to preserve the order but that can be done other way as in the updated diff.

Reply:
Yes. Its O(N) in both space and time which is the best we can go? Probably if we want to preserve space then we can sort the list in O(NlogN) by using O(logN) space (max size of recursion stack I think) but then it would be too complicated and slower.